src/drivers/general/queries/experimental/UnicodeStringFreed/UnicodeStringFreed.ql

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+/**
+ * @id cpp/drivers/unicode-string-not-freed
+ * @name Unicode String Not Freed
+ * @description UnicodeString objects created with RtlCreateUnicodeString must be freed with RtlFreeUnicodeString.
+ * @platform Desktop
+ * @security.severity Medium
+ * @feature.area Multiple
+ * @impact Exploitable Design
+ * @repro.text A UNICODE_STRING object is created with RtlCreateUnicodeString but not freed with RtlFreeUnicodeString.
+ * @owner.email sdat@microsoft.com
+ * @opaqueid CQLD-D0005
+ * @kind problem
+ * @problem.severity warning
+ * @precision medium
+ * @tags correctness
+ *       wddst
+ * @scope domainspecific
+ * @query-version v1
+ */
+
+
+import cpp
+import semmle.code.cpp.dataflow.new.DataFlow
+
+class UnicodeStringAccess extends VariableAccess {
```

Comment:
Nitpick: probably cleaner to define this as an access to a UnicodeString rather than repeating the .getType().getName() logic.